File: client/webui/src/main/webapp/hkt/js/service/KpiService.js
/**
 * 
*/
define([
	'jquery',
	'service/Server',
], function ($, Server){
	var KpiService= {
			getCityByCode: function(code){
				var request={
						module: 'kpi', service:'KpiService', method: 'getCityByCode',
						params: {code: code}
				};
				return Server.POST(request);
			},
			getCityByCodeCountry:function(codeCountry){
				var request= {
						model: 'kpi', service: 'KpiService', method:'getCityByCodeCountry',
						params: {codeCountry:codeCountry}
				};
				return Server.POST(request);
			},
			getAllCity : function() {
			      var request = { 
			        module : 'kpi', service : 'KpiService', method : 'getAllCity',
			        params : {}
			      };
			      return Server.POST(request);
			 },
			 saveCity : function(city) {
			      var request = {
			        module : 'kpi', service : 'KpiService', method : 'saveCity',
			        params : { city : city }
			      };
			      return Server.POST(request);
			 },
			 deleteCity : function(city) {
			      var request = {
			        module : 'kpi', service : 'KpiService', method : 'deleteCity',
			        params : { city : city }
			      };
			      return Server.POST(request);
			 },
			    
	};
	return KpiService;
}); 

File: module/kpi/service/src/main/java/com/hkt/module/kpi/KpiService.java
package com.hkt.module.kpi;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import com.hkt.module.kpi.entity.City;
import com.hkt.module.kpi.repository.CityRepository;

@Service("KpiService")
public class KpiService {

	@Autowired
	private CityRepository cityRepository;
	
	public KpiService() {
		super();
	}

	@Transactional(readOnly = true)
	public City getCityByCode(String code) {
		return cityRepository.getByCode(code);
	}

	@Transactional(readOnly = true)
	public List<City> getCityByCodeCountry(String codeCountry) {
		return cityRepository.getByCodeCountry(codeCountry);
	}

	@Transactional(readOnly = true)
	public List<City> getAllCity() {
		return (List<City>) cityRepository.findAll();
	}
	@Transactional
	public City saveCity(City city)
	{
		return cityRepository.save(city);
	}
	@Transactional
	public void deleteCity(City city)
	{
		System.out.println(city);
		cityRepository.delete(city);
	}

}
